docker/Dockerfile-pgtests

```diff
@@ -1,12 +1,21 @@
 # Use the Sytest image that comes with a lot of the build dependencies
 # pre-installed
-FROM matrixdotorg/sytest:latest
+FROM matrixdotorg/sytest:bionic
 
 # The Sytest image doesn't come with python, so install that
 RUN apt-get update && apt-get -qq install -y python3 python3-dev python3-pip
 
 # We need tox to run the tests in run_pg_tests.sh
 RUN python3 -m pip install tox
 
+# Initialise the db
+RUN su -c '/usr/lib/postgresql/10/bin/initdb -D /var/lib/postgresql/data -E "UTF-8" --lc-collate="en_US.UTF-8" --lc-ctype="en_US.UTF-8" --username=postgres' postgres
+
+# Create a volume for tox's workdir (so we don't have to fetch dependencies from PyPI every run)
+VOLUME /tox
+
 ADD run_pg_tests.sh /pg_tests.sh
-ENTRYPOINT /pg_tests.sh
+
+# Use the "exec form" of ENTRYPOINT (https://docs.docker.com/engine/reference/builder/#entrypoint)
+# so that we can `docker run` this container and pass arguments to pg_tests.sh
+ENTRYPOINT ["/pg_tests.sh"]
```

Comment:
Whilst you're here, do you think it'd be sensible to make it keep the same filename when it's copied into the container? It's a bit sad to have two different (context-dependent) names for the same script; I was about to look to see if there was another script called 'pg_tests.sh'.

Reply:
Very happy to do this!